python/ray/tune/sample.py

```diff
@@ -6,9 +6,62 @@
 
 import numpy as np
 
+# Backwards compatibility
+try:
+    # Added in numpy>=1.17 but we require numpy>=1.16
+    random_generator = np.random.Generator
+    LEGACY_RNG = False
+except AttributeError:
+
+    class random_generator:
+        pass
+
+    LEGACY_RNG = True
+
 logger = logging.getLogger(__name__)
 
 
+class _BackwardsCompatibleNumpyRng:
+    """Thin wrapper to ensure backwards compatibility between
+    new and old numpy randomness generators.
+    """
+    _rng = None
+
+    def __init__(self,
+                 generator_or_seed: Optional[Union[
+                     "random_generator", np.random.RandomState, int]] = None):
+        if generator_or_seed is None:
+            self._rng = None
+            return
+        if isinstance(generator_or_seed, np.random.RandomState):
+            self._rng = generator_or_seed
+            return
+        if isinstance(generator_or_seed, random_generator):
+            self._rng = generator_or_seed
+            return
+        if LEGACY_RNG:
+            self._rng = np.random.RandomState(generator_or_seed)
+        else:
+            self._rng = np.random.default_rng(generator_or_seed)
```

Comment:
```suggestion
        if generator_or_seed is None:
            self._rng = np.random
        elif isinstance(generator_or_seed, np.random.RandomState):
            self._rng = generator_or_seed
        elif isinstance(generator_or_seed, random_generator):
            self._rng = generator_or_seed
        elif LEGACY_RNG:
            self._rng = np.random.RandomState(generator_or_seed)
        else:
            self._rng = np.random.default_rng(generator_or_seed)
```

Reply:
By the way, doesn't this sometimes use the stub `random_generator` class?

Reply:
Unless the user explicitly imports the stub and tries to use it, this will always return false